### PR TITLE
Revamped Key Vault Administration Javadocs

### PR DESCRIPTION
**Pull request by vcolin7**

### CODE REVIEW

**Comment by vcolin7** [comment]
Merging blocked until https://github.com/Azure/autorest.java/issues/2321 is resolved.

We could also merge all the changes other than those for the `package-info.java`... What do you think @g2vinay?

File: sdk/keyvault/azure-security-keyvault-administration/src/main/java/com/azure/security/keyvault/administration/package-info.java
Comment (g2vinay) [inline sdk/keyvault/azure-security-keyvault-administration/src/main/java/com/azure/security/keyvault/administration/package-info.java:16]: Wrap the questions in `<strong>` tag in all places
<p><strong>What is a Key Vault Access Control Client?</strong></p>

File: sdk/keyvault/azure-security-keyvault-administration/src/main/java/com/azure/security/keyvault/administration/package-info.java
Comment (g2vinay) [inline sdk/keyvault/azure-security-keyvault-administration/src/main/java/com/azure/security/keyvault/administration/package-info.java:64]: It will be helpful to link out to a MS Docs document explaining these concepts as well.

Comment (vcolin7) [reply sdk/keyvault/azure-security-keyvault-administration/src/main/java/com/azure/security/keyvault/administration/package-info.java:64]: Do we do this for other libraries? Is it a document other than the README of the library? Because that's what end up published at MS Docs.

Comment (vcolin7) [reply sdk/keyvault/azure-security-keyvault-administration/src/main/java/com/azure/security/keyvault/administration/package-info.java:64]: Apparently, we don't have docs for this :(

File: sdk/keyvault/azure-security-keyvault-administration/src/main/java/com/azure/security/keyvault/administration/KeyVaultAccessControlAsyncClient.java
Comment (alzimmermsft) [inline sdk/keyvault/azure-security-keyvault-administration/src/main/java/com/azure/security/keyvault/administration/KeyVaultAccessControlAsyncClient.java:90]: I don't think the correct sample is being used here, this looks to be the synchronous sample. This appears to be an issue with multiple samples in this class.

File: sdk/keyvault/azure-security-keyvault-administration/src/main/java/com/azure/security/keyvault/administration/KeyVaultAccessControlAsyncClient.java
Comment (alzimmermsft) [inline sdk/keyvault/azure-security-keyvault-administration/src/main/java/com/azure/security/keyvault/administration/KeyVaultAccessControlAsyncClient.java:43]: I don't see any listing examples

File: sdk/keyvault/azure-security-keyvault-administration/src/main/java/com/azure/security/keyvault/administration/KeyVaultAccessControlClient.java
Comment (alzimmermsft) [inline sdk/keyvault/azure-security-keyvault-administration/src/main/java/com/azure/security/keyvault/administration/KeyVaultAccessControlClient.java:173]:
```suggestion
 * <h2>Get a Role Assignment</h2>
```

File: sdk/keyvault/azure-security-keyvault-administration/src/main/java/com/azure/security/keyvault/administration/KeyVaultAccessControlAsyncClient.java
Comment (alzimmermsft) [inline sdk/keyvault/azure-security-keyvault-administration/src/main/java/com/azure/security/keyvault/administration/KeyVaultAccessControlAsyncClient.java:173]:
```suggestion
 * <h2>Get a Role Assignment</h2>
```

File: sdk/keyvault/azure-security-keyvault-administration/src/main/java/com/azure/security/keyvault/administration/KeyVaultAccessControlClient.java
Comment (alzimmermsft) [inline sdk/keyvault/azure-security-keyvault-administration/src/main/java/com/azure/security/keyvault/administration/KeyVaultAccessControlClient.java:196]:
```suggestion
 * <h2>Delete a Role Assignment</h2>
```

File: sdk/keyvault/azure-security-keyvault-administration/src/main/java/com/azure/security/keyvault/administration/KeyVaultAccessControlClient.java
Comment (alzimmermsft) [inline sdk/keyvault/azure-security-keyvault-administration/src/main/java/com/azure/security/keyvault/administration/KeyVaultAccessControlClient.java:201]: Wrong API

File: sdk/keyvault/azure-security-keyvault-administration/src/main/java/com/azure/security/keyvault/administration/KeyVaultAccessControlAsyncClient.java
Comment (alzimmermsft) [inline sdk/keyvault/azure-security-keyvault-administration/src/main/java/com/azure/security/keyvault/administration/KeyVaultAccessControlAsyncClient.java:196]:
```suggestion
 * <h2>Delete a Role Assignment</h2>
```

File: sdk/keyvault/azure-security-keyvault-administration/src/main/java/com/azure/security/keyvault/administration/KeyVaultAccessControlAsyncClient.java
Comment (alzimmermsft) [inline sdk/keyvault/azure-security-keyvault-administration/src/main/java/com/azure/security/keyvault/administration/KeyVaultAccessControlAsyncClient.java:201]: Wrong API

File: sdk/keyvault/azure-security-keyvault-administration/src/main/java/com/azure/security/keyvault/administration/KeyVaultAccessControlClientBuilder.java
Comment (alzimmermsft) [inline sdk/keyvault/azure-security-keyvault-administration/src/main/java/com/azure/security/keyvault/administration/KeyVaultAccessControlClientBuilder.java:48]: I'd pluralize client

File: sdk/keyvault/azure-security-keyvault-administration/src/main/java/com/azure/security/keyvault/administration/KeyVaultAccessControlClientBuilder.java
Comment (alzimmermsft) [inline sdk/keyvault/azure-security-keyvault-administration/src/main/java/com/azure/security/keyvault/administration/KeyVaultAccessControlClientBuilder.java:66]:
```suggestion
 * {@link HttpClient HTTP client} can be optionally configured in the {@link KeyVaultAccessControlClientBuilder}.</p>
```

File: sdk/keyvault/azure-security-keyvault-administration/src/main/java/com/azure/security/keyvault/administration/KeyVaultAccessControlClientBuilder.java
Comment (alzimmermsft) [inline sdk/keyvault/azure-security-keyvault-administration/src/main/java/com/azure/security/keyvault/administration/KeyVaultAccessControlClientBuilder.java:65]: Should state its HTTP log detail level as it doesn't change how other logging is done. We're also setting HttpLogOptions and not just the log level in the sample

File: sdk/keyvault/azure-security-keyvault-administration/src/main/java/com/azure/security/keyvault/administration/KeyVaultAccessControlClientBuilder.java
Comment (alzimmermsft) [inline sdk/keyvault/azure-security-keyvault-administration/src/main/java/com/azure/security/keyvault/administration/KeyVaultAccessControlClientBuilder.java:79]: We duplicate this for building the sync client but don't duplicate the more advanced sample for the sync client

File: sdk/keyvault/azure-security-keyvault-administration/src/main/java/com/azure/security/keyvault/administration/KeyVaultAccessControlClientBuilder.java
Comment (alzimmermsft) [inline sdk/keyvault/azure-security-keyvault-administration/src/main/java/com/azure/security/keyvault/administration/KeyVaultAccessControlClientBuilder.java:85]: I see the vault URL was updated for the more complex example, does it need to be updated for the other examples as well?

File: sdk/keyvault/azure-security-keyvault-administration/src/main/java/com/azure/security/keyvault/administration/KeyVaultBackupAsyncClient.java
Comment (alzimmermsft) [inline sdk/keyvault/azure-security-keyvault-administration/src/main/java/com/azure/security/keyvault/administration/KeyVaultBackupAsyncClient.java:47]:
```suggestion
 * The {@link KeyVaultBackupAsyncClient} provides asynchronous methods to perform full backup and restore of a key
```

File: sdk/keyvault/azure-security-keyvault-administration/src/main/java/com/azure/security/keyvault/administration/KeyVaultBackupClient.java
Comment (alzimmermsft) [inline sdk/keyvault/azure-security-keyvault-administration/src/main/java/com/azure/security/keyvault/administration/KeyVaultBackupClient.java:50]:
```suggestion
 * The {@link KeyVaultBackupClient} provides synchronous methods to perform full backup and restore of a key vault,
```

File: sdk/keyvault/azure-security-keyvault-administration/src/main/java/com/azure/security/keyvault/administration/package-info.java
Comment (alzimmermsft) [inline sdk/keyvault/azure-security-keyvault-administration/src/main/java/com/azure/security/keyvault/administration/package-info.java:44]: This section is worded slightly different. Does the Key Vault Backup Client have both sync and async operations in both the sync and async clients or is this referring to the existence of the sync and async clients?

**Review by g2vinay — APPROVED** [review]
Good to merge, post addressing pending comments.

**Pull request merged**
